Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -912,12 +923,17 @@ def resource_version(
         puts if debug && strategy.blank?
         next if strategy.blank?
 
-        strategy_data = strategy.find_versions(
+        strategy_args = {
           url:           url,
-          regex:         livecheck_regex,
           homebrew_curl: false,
-          &livecheck_strategy_block
-        )
+        }
+        strategy_args[:regex] = livecheck_regex if livecheck_regex.present?
```

Comment:
Leave this in the hash and call `compact` on it.

Reply:
Oh yeah, that's much nicer. I somehow forgot that `#compact` also works for hashes.